pkg/antctl/raw/multicluster/init.go

```diff
@@ -0,0 +1,168 @@
+`, "\n")
+
+func NewInitCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:     "init",
+		Short:   "Initialize ClusterSet in the given Namespace of the leader cluster",
+		Args:    cobra.MaximumNArgs(0),
+		Example: initExample,
+		RunE:    initRunE,
+	}
+
+	o := initOptions{}
+	initOpts = &o
+	command.Flags().StringVarP(&o.namespace, "namespace", "n", "", "Namespace of the ClusterSet")
+	command.Flags().StringVarP(&o.clusterSet, "clusterset", "", "", "ClusterSet ID of the leader cluster")
+	command.Flags().StringVarP(&o.clusterID, "clusterid", "", "", "ClusterID of the leader cluster")
+	command.Flags().BoolVarP(&o.createToken, "create-token", "", false, "If specified, a default member token will be created")
+	command.Flags().StringVarP(&o.output, "output-file", "o", "", "Output file to save the member cluster join config")
+
+	return command
+}
+
+func initRunE(cmd *cobra.Command, args []string) error {
+	if err := initOpts.validate(); err != nil {
+		return err
+	}
+	k8sClient, err := common.NewClient(cmd)
+	if err != nil {
+		return err
+	}
+	createdRes := []map[string]interface{}{}
+	var createErr error
+	defer func() {
+		if createErr != nil {
+			fmt.Fprintf(cmd.OutOrStderr(), "Failed to init the Antrea Multi-cluster. Deleting the created resources\n")
+			if err := common.Rollback(cmd, k8sClient, createdRes); err != nil {
+				fmt.Fprintf(cmd.OutOrStdout(), "Failed to rollback: %v\n", err)
+			}
+		}
+	}()
+	createErr = common.CreateClusterClaim(cmd, k8sClient, initOpts.namespace, initOpts.clusterSet, initOpts.clusterID, &createdRes)
+	if createErr != nil {
+		return createErr
+	}
+	createErr = common.CreateClusterSet(cmd, k8sClient, initOpts.namespace, initOpts.clusterSet, "", "", "", initOpts.clusterID, initOpts.namespace, &createdRes)
+	if createErr != nil {
+		return createErr
+	}
+
+	var file *os.File
+	if initOpts.output != "" {
+		if file, err = os.OpenFile(initOpts.output, os.O_WRONLY|os.O_CREATE|os.O_TRUNC|os.O_APPEND, 0644); err != nil {
+			fmt.Fprintf(cmd.OutOrStderr(), "Failed to open file %s: %v\n", initOpts.output, err)
+		}
+	}
+	defer file.Close()
+
+	if err := outputConfig(cmd, file); err != nil {
+		return err
+	}
+	if initOpts.createToken {
+		if createErr = common.CreateMemberToken(cmd, k8sClient, defaultToken, initOpts.namespace, file, &createdRes); createErr != nil {
+			fmt.Fprintf(cmd.OutOrStderr(), "Failed to create Secret: %v\n", createErr)
+			return createErr
+		}
+	}
+
+	fmt.Fprintf(cmd.OutOrStdout(), "Successfully initialized ClusterSet %s\n", initOpts.clusterSet)
+
+	return nil
+}
+
+func outputConfig(cmd *cobra.Command, file *os.File) error {
+	if file == nil {
+		return nil
+	}
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+
+	config := &ClusterSetJoinConfig{
+		APIVersion:      common.ClusterSetJoinConfigAPIVersion,
+		Kind:            common.ClusterSetJoinConfigKind,
+		Namespace:       "",
```

Comment:
@hjiajing : when we unmarshal the YAML, will these "" fields overwrite values in ClusterSetJoinConfig?

Reply:
If so, I suggest not to set these fields in the generated file, as I hope the `join` command can use command line options to provide these arguments which seems the most convenient way for users.

I meant something like `antctl mc join --clusterid test-cluster-leader -n kube-system --config-file join-config.yml`

Or even better, if we can write `clusterID` and `namespace` and other optional fields to the config file as comments:
```
#clusterID: ""
#namespace: ""
# Use the pre-created token Secret.
#tokenSecretName: ""
# Create a token Secret with the manifest file.
#toeknSecretFile: ""
```

 See also the discussion at https://github.com/antrea-io/antrea/pull/4096#discussion_r942004768.

Reply:
Added a raw string constant now the output join config looks like 

```yaml
---
kind: ClusterSetJoinConfig
apiVersion: multicluster.antrea.io/v1alpha1
clusterSetID: test-clusterset
leaderClusterID: test-cluster-north
leaderNamespace: antrea-multicluster
leaderAPIServer: https://172.18.0.3:6443
clusterID: "test-cluster-east"
#namespace: ""
# Use the pre-created token Secret.
#tokenSecretName: ""
# Create a token Secret with the manifest file.
#toeknSecretFile: ""
```

Reply:
I feel we should comment out clusterID too, so it can be overwritten by command line argument.

#clusterID: ""

Reply:
Sorry I copied the output file which I edited. The output is 

```yaml
"config.yaml" 25L, 3372B                                                                                                                  13,1          All
---
kind: ClusterSetJoinConfig
apiVersion: multicluster.antrea.io/v1alpha1
clusterSetID: test-clusterset
leaderClusterID: test-cluster-north
leaderNamespace: antrea-multicluster
leaderAPIServer: https://172.18.0.3:6443
#clusterID: ""
#namespace: ""
# Use the pre-created token Secret.
#tokenSecretName: ""
# Create a token Secret with the manifest file.
#toeknSecretFile: ""
```

The `clusterID` is commented out.